src/components/ProposalListTable.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {
+  Table,
+  TableHead,
+  ColumnHead,
+  TableBody,
+  TableRow,
+  RowCell,
+} from './Table';
+
+interface ProposalListTableRowProps {
+  // The `fields` array is currently required, as it determines the column order.
+  // If we can instead recompose the `proposal` children to be in the desired order,
+  // we don't need `fields`.
+  fields: {
+    shortCode: string,
+    label: string,
+  }[],
+  // Note the subtle difference from `proposals` on `ProposalListTableProps`:
+  // this is a single proposal, not an array of them.
+  proposal: {
+    fieldName: string,
+    value: string,
+  }[],
+  proposalId: string,
+}
+
+const ProposalListTableRow = ({
+  fields,
+  proposal,
+  proposalId,
+}: ProposalListTableRowProps) => {
+  const handleRowClick = () => {
+    // TODO: Replace this with a navigation to the proposal detail page.
+    // eslint-disable-next-line no-console
+    console.log(`Navigate to proposal detail page for ${proposalId}`);
+  };
+
+  return (
+    <TableRow onClick={handleRowClick}>
+      {/*
+        Iterate over fields to ensure order. Alternatively, if we can guarantee the proposal
+        field order matches the order of `fields`, we can do away with this and just iterate
+        over proposal fields sequentially.
+      */}
+      {fields.map((field) => {
+        // Note the two assumptions baked in here (also mentioned in the props design above):
+        // 1. Proposal `fieldName` corresponds to field `shortCode`
+        // 2. Each proposal has only one value for each matching field
```

Comment:
I think we'll want to assume there can be multiple (or zero) values for a given field, and just always expect a list.

Reply:
@jasonaowen lol, yes, absolutely. I clumsily was trying to say "here are the two _incorrect_ assumptions baked in to this implementation". Your suggested format above, with values as an array, is great and easily resolves this limitation.